src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/listeners/CollectFailedTestLogsListener.groovy

```diff
@@ -0,0 +1,52 @@
+package org.openkilda.functionaltests.listeners
+
+import com.fasterxml.jackson.databind.ObjectMapper
+import org.spockframework.runtime.model.ErrorInfo
+import org.spockframework.runtime.model.IterationInfo
+import org.springframework.beans.factory.annotation.Value
+import org.springframework.web.client.RestTemplate
+
+import java.time.ZoneOffset
+import java.util.concurrent.ConcurrentHashMap
+
+import static java.time.LocalDateTime.now
+import static java.time.format.DateTimeFormatter.ISO_DATE_TIME
+
+class CollectFailedTestLogsListener extends AbstractSpringListener{
+    private ConcurrentHashMap<String, String> startTime = new ConcurrentHashMap<>()
+    @Value('${elasticsearch.endpoint}')
+    private String elasticSearchEndpoint
+    @Value('${elasticsearch.index}')
+    private String elasticSearchIndex
+    private final static COMMON_TEST_PACKAGE_PATH = "org.openkilda.functionaltests.spec."
+
+
+    @Override
+    void beforeIteration(IterationInfo iterationInfo) {
+        startTime.put(iterationInfo.getDisplayName(), utcTimeNow())
+    }
+
+    @Override
+    void error(ErrorInfo error) {
+        def objectMapper = new ObjectMapper()
+        def startTime = startTime.get(error.getMethod().getIteration().getDisplayName())
+        def logs = new RestTemplate().getForEntity(
+                "${elasticSearchEndpoint}${elasticSearchIndex}/_search?q=" +
+                        "@timestamp:[${startTime} TO ${utcTimeNow()}]", String.class).getBody()
+        def beautifiedString = objectMapper.readValue(logs, Object.class)
+        objectMapper.writerWithDefaultPrettyPrinter().writeValue(getTargetLogFile(error), beautifiedString)
+    }
+
+    private static String utcTimeNow() {
+        return now(ZoneOffset.UTC).format(ISO_DATE_TIME)
+    }
+
+    private static File getTargetLogFile(ErrorInfo error) {
```

Comment:
It'd be a good idea for this and https://github.com/telstra/open-kilda/blob/558e34f3f9476aa0f83b024657ff982d862271dc/src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/listeners/LogParallelSpecsListener.groovy#L43
to find 'logs directory' for current test using some common shared method